cmd/install.go

```diff
@@ -90,8 +102,67 @@ Provide the --local-path flag with --merge if a kubeconfig already exists in som
 			k3sExtraArgs += `--no-deploy servicelb --no-deploy traefik`
 		}
 
-		installk3sExec := fmt.Sprintf("INSTALL_K3S_EXEC='server %s --tls-san %s %s'", clusterStr, ip, strings.TrimSpace(k3sExtraArgs))
+		installk3sExec := ""
+		if ds != "" {
+			databaseUser, _ := command.Flags().GetString("db-username")
+			databasePassword, _ := command.Flags().GetString("db-passwd")
+			databaseHostname, _ := command.Flags().GetString("db-hostname")
+			databasePort, _ := command.Flags().GetString("db-port")
+			databaseName, _ := command.Flags().GetString("db-name")
+
+			if ds == "sql" {
+				if databaseUser == "" {
+					databaseUser = "root"
+				}
+
+				if databasePassword == "" {
+					return fmt.Errorf("provide password for mysql user")
+				}
+
+				if databaseHostname == "" {
+					databaseHostname = "/var/run/mysqld/mysqld.sock"
+				}
+
+				if databasePort == "" {
+					databasePort = "3306"
+				}
+
+				if databaseName == "" {
+					databaseName = "kubernetes"
+				}
+
+				datastoreStr = fmt.Sprintf("mysql://%s:%s@tcp(%s:%s)/%s", databaseUser, databasePassword, databaseHostname, databasePort, databaseName)
```

Comment:
Sorry that I missed this before. We can make things much simpler I think, just by accepting a connection string.